client/src/graphql_utils/graphql_utils.js

```diff
@@ -97,70 +106,98 @@ export function get_client() {
-        log_standard_event({
-          SUBAPP: window.location.hash.replace("#", ""),
-          MISC1: "API_QUERY_SUCCESS",
-          MISC2: `${query_name}, took ${resp_time} ms`,
-        });
-      } else {
-        log_standard_event({
-          SUBAPP: window.location.hash.replace("#", ""),
-          MISC1: "API_QUERY_UNEXPECTED",
-          MISC2: `${query_name}, took ${resp_time} ms`,
-        });
-      }
+      log_standard_event({
+        SUBAPP: window.location.hash.replace("#", ""),
+        MISC1: "API_QUERY_SUCCESS",
+        MISC2: `${query_name}, took ${Date.now() - time_at_request} ms`,
+      });
 
       return resolved_response;
     })
     .catch((error) => {
-      const resp_time = Date.now() - time_at_request;
       log_standard_event({
         SUBAPP: window.location.hash.replace("#", ""),
         MISC1: "API_QUERY_FAILURE",
-        MISC2: `${query_name}, took ${resp_time} ms - ${error.toString()}`,
+        MISC2: `${query_name}, took ${
+          Date.now() - time_at_request
+        } ms - ${error.toString()}`,
       });
+
       throw error;
     });
 };
 
-const InnerLoadingHoc = ({ Component, data_to_props }) => (props) => {
-  if (props.data.loading) {
-    return <div> Loading ... </div>;
+const make_query_hook = (query_name, query, resolver) => (variables) => {
+  const [time_at_request, set_time_at_request] = useState(Date.now()); // eslint-disable-line no-unused-vars
+
+  const { loading, error, data } = useQuery(query, {
+    variables: {
+      ...variables,
+      _query_name: query_name,
+    },
+  });
+
+  if (loading) {
+    return {
+      loading,
+      error,
+      data,
+    };
+  } else if (error) {
+    log_standard_event({
+      SUBAPP: window.location.hash.replace("#", ""),
+      MISC1: "API_QUERY_FAILURE",
+      MISC2: `${query_name}, took ${
+        Date.now() - time_at_request
+      } ms - ${error.toString()}`,
+    });
+
+    throw new Error(error);
   } else {
-    return (
-      <Component
-        data={data_to_props(props.data)}
-        gql_props={{
-          refetch: props.data.refetch,
-          variables: props.data.variables,
-        }}
-      />
-    );
+    log_standard_event({
+      SUBAPP: window.location.hash.replace("#", ""),
+      MISC1: "API_QUERY_SUCCESS",
+      MISC2: `${query_name}, took ${Date.now() - time_at_request} ms`,
+    });
+
+    return {
+      loading,
+      error,
+      data: resolver(data),
+    };
   }
 };
 
-//for use in development only
-export const LoadingHoc = ({
-  Component,
-  query,
-  data_to_props = _.identity,
-  variables,
-}) =>
-  apollo_connect(query, {
-    options: variables ? { variables } : {},
-  })(InnerLoadingHoc({ Component, data_to_props }));
-
-assign_to_dev_helper_namespace({
-  query_api: (query) => client.query({ query }),
-});
+export const query_maker = ({ query_name, query, resolver = _.identity }) => {
```

Comment:
this is also a hook, and should use the `use*` naming convention

Reply:
Same as above. `query_maker` returns `{query_<query_name>, use<query_name, in pascal case>}`. The first being a promise based query, the second being the hook equivalent.